lib/spack/spack/util/system.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+"""
+Utility methods for interfacing with various system utilities
+"""
```

Comment:
IMO `system` is too generic a name. We call our Spack augmented YAML handler `spack_yaml`, and I think that would be appropriate to emulate here, like `spack _filesystem`.